Plugin/MapStringScopeToInt.php

```diff
@@ -21,17 +21,16 @@ class MapStringScopeIdToInt
      * about stemming and synonyms as we are literally just using it for the Lucene Query Syntax we can safely ignore
      * this and remap and provide a fake store view id of 0.
      *
-     * @see vendor/magento/module-elasticsearch/Model/Adapter/Index/Builder.php:63
-     * @param SynonymReader $subject
-     * @param int|string $storeViewId
+     * @param BuilderInterface $subject
+     * @param int $storeId
```

Comment:
@gowrizrh, is this still accurate? Can we restrict the input type? Do we need to check `is_string` on this variable still?

Reply:
Good catch. It must be `int|string`. Unfortunately we can't restrict the input type. 

Magento really just sends it as a string (but a numeric) and downstream it just casts it (legacy code). Therefore this plugin is needed to check if it's a non numerical string then just provide a dummy value.

Reply:
All good, revoked my approval so you'd catch the comment but obviously safe to merge :ok_hand: